### PR TITLE
feat: api normalisation formElements and values

### PR DESCRIPTION
**Pull request by tlouisse**
## Changes

**Simplify / normalize api**
- Discriminate between just 3 types of form controls:
   1. **field**: regular (text) inputs like `lion-input`, `lion-textarea`, `lion-input-range`, `lion-input-(email|amount|date|datepicker)` etc. 
    -> _Children: none_
   2. **choice group**: like `radio-group`, `checkbox-group`, `select-rich`, (and todo: `lion-listbox`, `lion-combobox`)
    -> _Children: array of options_  
   3. **(sub) form**: those are `lion-form` and `lion-fieldset` and can be accessed both like an object and an array (just like platform). A (sub) form can be serialized as a group
    -> _Children: array of controls_, which will be either of category 1 or 2. Those children can be accessed by name
- formElements work like the platform `form.elements`. See: https://github.com/ing-bank/lion/blob/72fcd3768f9f59bfc21cc9ba90b21ddcf600cf74/packages/field/src/registration/FormControlsCollection.js
- modelValue/serializedValue are derived from formElements structure. The only exception is the ChoiceGroup, where only checked values are filtered out and a `multipleChoice=false` results in a single value
- `.serializedValue` and `serializeGroup()` have same outcome (except for disabled values being ignored). 
- `serializeGroup()` has two options: `keepDisabled` and `includeSelf`:  https://github.com/ing-bank/lion/blob/72fcd3768f9f59bfc21cc9ba90b21ddcf600cf74/packages/fieldset/test/lion-fieldset.test.js#L789. The latter addresses: https://github.com/ing-bank/lion/issues/399

**Fixes:**
- compute interaction state after registration: https://github.com/ing-bank/lion/blob/fix/serializedValues/packages/fieldset/test/lion-fieldset.test.js#L601

Cleanup:
- Removed FieldCustomMixin. It doesn't belong in `packages/field` which is actually the form core package. It has specific logic for our extension layer and should be placed there

### CODE REVIEW

**Comment by daKmoR** [comment]
Submit & Reset Demo no longer works :(

https://deploy-preview-583--lion-web-components.netlify.com/?path=/docs/forms-form-examples--default-story

File: packages/choice-input/src/ChoiceGroupMixin.js [hunk 1]
```diff
@@ -119,24 +143,25 @@ export const ChoiceGroupMixin = dedupeMixin(
...
 
       _getCheckedElements() {
-        const filtered = this.formElementsArray.filter(el => el.checked === true);
-        if (this.multipleChoice) {
-          return filtered;
-        }
-        return filtered.length > 0 ? filtered[0] : undefined;
+        // We want to filter out disabled values out by default
+        return this.formElements.filter(el => el.checked && !el.disabled);
+        // if (this.multipleChoice) {
```
Comment (daKmoR) [inline packages/choice-input/src/ChoiceGroupMixin.js]: why leave commented out code in there? imho should be removed

Comment (tlouisse) [reply packages/choice-input/src/ChoiceGroupMixin.js]: Definitely :) Alex pointed this already out as well. I'm on it

**Comment by tlouisse** [comment]
> Submit & Reset Demo no longer works :(
> 
> https://deploy-preview-583--lion-web-components.netlify.com/?path=/docs/forms-form-examples--default-story

I reverted the demo functionality to how it was before

**Pull request merged**